### PR TITLE
Pia 2966 injecting bottom bar

### PR DESCRIPTION
**Pull request by AdnanYupi**
PIA 2966 

Bottom bar injected into review screen:
Note: check injection of loading indicator into review bottom bar

### CODE REVIEW

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/review/multipage/view/ReviewNavigationBarBottomAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import android.content.DialogInterface.OnClickListener
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import net.gini.android.capture.databinding.GcReviewNavigationBarBottomBinding
+import net.gini.android.capture.view.CustomLoadingIndicatorAdapter
+import net.gini.android.capture.view.DefaultLoadingIndicatorAdapter
+import net.gini.android.capture.view.InjectedViewAdapter
+
+interface ReviewNavigationBarBottomAdapter: InjectedViewAdapter {
```
Comment (a-szotyori) [inline capture-sdk/sdk/src/main/java/net/gini/android/capture/review/multipage/view/ReviewNavigationBarBottomAdapter.kt:12]: [Suggestion] We can make this interface more abstract. We don't have to tell clients how to show the loading indicator. We can simply declare two methods: `showLoadingIndicator()` and `hideLoadingIndicator()`. In the default implementation we can then use the `OnButtonLoadingIndicatorAdapter` (oh, we have a dependency on PR #117).

Also renaming `onButtonStatus()` to `setContinueButtonEnabled(enabled: Boolean)` makes it clearer what should be done when implementing the interface.

Generally the interface method names should follow the imperative mood to tell clients what we want them to do in each method.

In summary here is my proposal for the interface:
```
interface ReviewNavigationBarBottomAdapter: InjectedViewAdapter {

    fun setOnContinueButtonClickListener(clickListener: View.OnClickListener)

    fun setOnAddPageButtonClickListener(clickListener: View.OnClickListener)

    fun setAddPageButtonVisibility(visibility: Int)

    fun setContinueButtonEnabled(enabled: Boolean)

    fun showLoadingIndicator()

    fun hideLoadingIndicator()
}
```

**Review by a-szotyori — COMMENTED** [review]
I think we need to refine the adapter a bit to let clients show loading indicators any way they want to.

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/review/multipage/view/ReviewNavigationBarBottomAdapter.kt [hunk 2]
```diff
@@ -4,30 +4,32 @@ import android.content.DialogInterface.OnClickListener
...
 import net.gini.android.capture.view.DefaultLoadingIndicatorAdapter
 import net.gini.android.capture.view.InjectedViewAdapter
+import net.gini.android.capture.view.OnButtonLoadingIndicatorAdapter
 
-interface ReviewNavigationBarBottomAdapter: InjectedViewAdapter {
+interface ReviewNavigationBarBottomAdapter : InjectedViewAdapter {
 
     fun onContinueClickListener(clickListener: View.OnClickListener)
 
     fun onAddPageClickListener(clickListener: View.OnClickListener)
```
Comment (a-szotyori) [inline capture-sdk/sdk/src/main/java/net/gini/android/capture/review/multipage/view/ReviewNavigationBarBottomAdapter.kt:18]: [Request] Please rename to `setOnAddPageButtonClickListener`.

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/review/multipage/view/ReviewNavigationBarBottomAdapter.kt [hunk 3]
```diff
@@ -4,30 +4,32 @@ import android.content.DialogInterface.OnClickListener
...
 
     fun onAddPageVisible(visibility: Int)
```
Comment (a-szotyori) [inline capture-sdk/sdk/src/main/java/net/gini/android/capture/review/multipage/view/ReviewNavigationBarBottomAdapter.kt:20]: [Request] Please rename to `setAddPageButtonVisibility`.

**Review by a-szotyori — CHANGES_REQUESTED** [review]
Thank you for the changes! Sorry for being picky with the method names. They are part of the SDK's public API and we need to follow the same naming conventions in all the adapter interfaces.

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/review/multipage/view/ReviewNavigationBarBottomAdapter.kt [hunk 4]
```diff
@@ -4,30 +4,32 @@ import android.content.DialogInterface.OnClickListener
...
 import net.gini.android.capture.databinding.GcReviewNavigationBarBottomBinding
 import net.gini.android.capture.view.CustomLoadingIndicatorAdapter
 import net.gini.android.capture.view.DefaultLoadingIndicatorAdapter
 import net.gini.android.capture.view.InjectedViewAdapter
+import net.gini.android.capture.view.OnButtonLoadingIndicatorAdapter
 
-interface ReviewNavigationBarBottomAdapter: InjectedViewAdapter {
+interface ReviewNavigationBarBottomAdapter : InjectedViewAdapter {
 
     fun onContinueClickListener(clickListener: View.OnClickListener)
```
[the remark this answers is not in the document]
Comment (AdnanYupi) [reply capture-sdk/sdk/src/main/java/net/gini/android/capture/review/multipage/view/ReviewNavigationBarBottomAdapter.kt:16]: No worries all good 👍

**Review by a-szotyori — APPROVED** [review]
Thank you!
